sim/src/test/scala/midasexamples/TutorialSuite.scala

```diff
@@ -192,6 +192,45 @@ class ParityVitisTest extends TutorialSuite("Parity", platformConfigs = classOf[
   runTest("verilator", true)
   runTest("vcs", true)
 }
+
+
+/** Trait so that we have a uniform numbering scheme for the plusargs tests
+  */
+trait PlusargsKey {
+  def getKey(groupNumber: Int, testNumber: Int): String = {
+    val key = (groupNumber << 4 | testNumber)
+    s"+plusargs_test_key=${key}"
+  }
+}
+
+class PlusargsGroup0 extends TutorialSuite("PlusargsModule", "PlusargsModuleTestConfigGroup0") with PlusargsKey {
+  it should "provide the correct default value, 3 slice" in {
+    assert(run("verilator", true, args = Seq(getKey(0,0))) == 0)
+  }
+
+  it should "accept an int from the command line" in {
+    assert(run("verilator", true, args = Seq(s"+plusar_v=3", getKey(0,1))) == 0)
+    assert(run("verilator", true, args = Seq(s"+plusar_v=${BigInt("f00000000", 16)}", getKey(0,2))) == 0)
+    assert(run("verilator", true, args = Seq(s"+plusar_v=${BigInt("f0000000000000000", 16)}", getKey(0,3))) == 0)
+  }
+
+  it should "reject large runtime values" in {
+    assert(run("verilator", true, args = Seq(s"+plusar_v=${BigInt("ff0000000000000000", 16)}", getKey(0,4))) != 0)
+  }
+}
+
+class PlusargsGroup1 extends TutorialSuite("PlusargsModule", "PlusargsModuleTestConfigGroup1") with PlusargsKey {
```

Comment:
If you could regroup the tests such that you could give each a more suggestive name (i.e., explaining what they are testing) that might help us rapidly debug stuff if/when these fail in CI. e.g. PlusArgBoolean vs PlusArgMultibit etc...